include/picongpu/simulationControl/MovingWindow.hpp

```diff
@@ -181,34 +186,55 @@ class MovingWindow
     void incrementSlideCounter(const uint32_t currentStep)
     {
         // do not slide twice in one simulation step
-        if (slidingWindowActive==true && lastSlideStep < currentStep)
+        if (currentStep < endSlidingOnStep && lastSlideStep < currentStep)
```

Comment:
since `endSlidingOnStep` defaults to `0u` and `currentStep` is `>=0` this is fine. But by the logic that was here before, you should first check if `isEnabled()` is true.

Reply:
Sounds good than it is more cleare